homeassistant/components/light/mqtt_json.py

```diff
@@ -307,20 +309,22 @@ def async_turn_on(self, **kwargs):
 
         message = {'state': 'ON'}
 
-        if ATTR_HS_COLOR in kwargs:
+        if ATTR_HS_COLOR in kwargs and (self._rgb or self._xy):
             hs_color = kwargs[ATTR_HS_COLOR]
-            brightness = kwargs.get(
-                ATTR_BRIGHTNESS, self._brightness if self._brightness else 255)
-            rgb = color_util.color_hsv_to_RGB(
-                hs_color[0], hs_color[1], brightness / 255 * 100)
-            xy_color = color_util.color_hs_to_xy(*kwargs[ATTR_HS_COLOR])
-            message['color'] = {
-                'r': rgb[0],
-                'g': rgb[1],
-                'b': rgb[2],
-                'x': xy_color[0],
-                'y': xy_color[1],
-            }
+            message['color'] = {}
+            if self._rgb:
+                brightness = kwargs.get(
+                    ATTR_BRIGHTNESS,
+                    self._brightness if self._brightness else 255)
```

Comment:
What if `self._brightness` is `0`? Then the hsv to rgb converter would check with full brightness. Probably not really a problem since when brightness is zero, rgb is essentially ignored by the remote device. But still this behavior of having very different values with `0` and `1` brightness seems weird to me. Might very well be wrong though

Reply:
Don't disagree, but I'd leave that up to someone who uses this platform. Probably checking for None was the original intention.

Reply:
Don't disagree, but I'd leave that up to someone who uses this platform. Probably checking for None was the original intention.

Reply:
I can confirm that PR fixes the problem with scenes